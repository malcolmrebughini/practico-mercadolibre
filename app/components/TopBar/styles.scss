@import 'app/styles/colors';
@import 'app/styles/vars';

$search-button-width: 46px;

.topBar {
  width: 100%;
  height: 54px;
  background-color: $ml-yellow;
  border-bottom: 1px solid rgba(51, 51, 51, 0.1);
}

.container {
  height: 54px;

  @media (min-width: 768px) {
    max-width: $body-width;
    margin: 0 auto;
  }

  .logo {
    display: table-cell;
    width: 58px;
    height: 40px;
    padding: 7px 5px;

    & > img {
      width: 58px;
      height: 40px;
    }
  }

  .search {
    display: table-cell;
    vertical-align: top;
    padding: 7px 7px 7px 5px;
    width: 100%;
    height: 54px;
    position: relative;
    box-sizing: border-box;

    & > form {
      display: table;
      width: 100%;

      & > input {
        display: table-cell;
        box-sizing: border-box;
        border: none;
        padding: 5px 30px 5px 15px;
        border-radius: 3px 0 0 3px;
        width: calc(100% - #{$search-button-width});
        height: 40px;
        vertical-align: middle;
        font-size: 18px;

        @media (min-width: 768px) {
          padding: 5px 60px 5px 15px;
        }
      }

      & > button {
        display: table-cell;
        background: url('/assets/magnifiying_glass.png');
        background-size: 18px 18px;
        background-position: center;
        background-repeat: no-repeat;
        width: $search-button-width;
        height: 40px;
        border: none;
        background-color: $ml-lighter-grey;
        border-radius: 0 3px 3px 0;
        vertical-align: middle;
      }
    }
  }
}

