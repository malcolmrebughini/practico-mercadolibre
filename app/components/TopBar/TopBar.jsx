import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import mlLogo from './assets/ml_logo.png';
import styles from './styles.scss';


class TopBar extends React.Component {
  onInputChange = (evt) => {
    const value = evt.target.value;
    this.props.updateInput(value);
  }

  onFormSubmit = (evt) => {
    evt.preventDefault();
    this.props.redirectToResultsView(this.props.inputValue);
  }

  render = () => {
    return (
      <div className={styles.topBar}>
        <div className={styles.container}>
          <Link className={styles.logo} to={{ pathname: '/' }}>
            <img src={mlLogo} />
          </Link>
          <div className={styles.search}>
            <form onSubmit={this.onFormSubmit}>
              <input
                className={styles.searchInput}
                value={this.props.inputValue}
                onChange={this.onInputChange}
              />
              <button className={styles.searchButton} type="submit"></button>
            </form>
          </div>
        </div>
      </div>
    );
  }
};


TopBar.propTypes = {
  search: PropTypes.func,
};


export default TopBar;
