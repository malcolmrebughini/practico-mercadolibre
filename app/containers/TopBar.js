import { connect } from 'react-redux';
import actions from '../actions';
import TopBar from '../components/TopBar';


function mapStateToProps(state) {
  return {
    inputValue: state.topBar.inputValue,
  };
}

export default connect(mapStateToProps, actions)(TopBar);
