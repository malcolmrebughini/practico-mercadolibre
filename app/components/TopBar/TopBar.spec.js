import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TopBar from './TopBar';


describe('TopBar', () => {
  it('should render correctly', () => {
    const mockCallbackUpdate = jest.fn();
    const mockCallbackRedirect = jest.fn();
    const inputValue = 'Mala fama';
    const wrapper = shallow(
      <TopBar
        updateInput={mockCallbackUpdate}
        redirectToResultsView={mockCallbackRedirect}
        inputValue={inputValue}
      />
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should call function on input', () => {
    const mockCallback = jest.fn();
    const wrapper = shallow(<TopBar updateInput={mockCallback} />);
    wrapper.find('input').simulate('change', { target: { value: 'mala fama' } });
    const calls = mockCallback.mock.calls;
    expect(calls.length).toEqual(1);
    expect(calls[0]).toEqual(['mala fama']);
  });

  it('should call function on submit', () => {
    const mockCallback = jest.fn();
    const wrapper = shallow(
      <TopBar
        inputValue="mala fama"
        redirectToResultsView={mockCallback}
      />
    );
    wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
    const calls = mockCallback.mock.calls;
    expect(calls.length).toEqual(1);
    expect(calls[0]).toEqual(['mala fama']);
  });
});
