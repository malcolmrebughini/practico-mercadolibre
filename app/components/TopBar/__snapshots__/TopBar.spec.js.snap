// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopBar should render correctly 1`] = `
<div
  className="topBar"
>
  <div
    className="container"
  >
    <Link
      className="logo"
      replace={false}
      to={
        Object {
          "pathname": "/",
        }
      }
    >
      <img
        alt="mercadolibre"
        src="some-static-file"
      />
    </Link>
    <div
      className="search"
    >
      <form
        action="/items"
        method="GET"
        onSubmit={[Function]}
      >
        <input
          className="searchInput"
          name="q"
          onChange={[Function]}
          value="Mala fama"
        />
        <button
          className="searchButton"
          type="submit"
        />
      </form>
    </div>
  </div>
</div>
`;
